packages/relay/src/lib/eth.ts

```diff
@@ -1737,7 +1737,11 @@ export class EthImpl implements Eth {
     }
 
     const cacheKeySyntheticLog = `${constants.CACHE_KEY.SYNTHETIC_LOG_TRANSACTION_HASH}${hash}`;
-    const cachedLog = this.cacheService.get(cacheKeySyntheticLog, EthImpl.ethGetTransactionReceipt, requestIdPrefix);
+    const cachedLog = await this.cacheService.getSharedWithFallback(
```

Comment:
This will never get any value from the fallback for the internal cache currently. Because when we use `set` with `shared == true`, and for some reason it fails to actually save the value to the cache, it does not go and saves it in the internal cache.

Reply:
I was under the impression that it does.
https://github.com/hashgraph/hedera-json-rpc-relay/blob/main/packages/relay/src/lib/services/cacheService/cacheService.ts#L198

You are right for failures inside the try...catch.

But this is just for when the `sharedCache` is not enabled.


```
if (shared && this.isSharedCacheEnabled)
```

for the above condition, `shared` is true, but `this.isSharedCacheEnabled` is false. it will set it on the internal cache.